cfme/scripting/report_parser.py

```diff
@@ -0,0 +1,97 @@
+"""Script to parse test reports."""
+import json
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+import click
+import yaml
+
+from cfme.utils.log import add_stdout_handler
+from cfme.utils.log import logger
+
+# Log to stdout too
+add_stdout_handler(logger)
+
+
+class ReportParser:
+    def __init__(self, report_path, report_format='json', report_dict=None, *args, **kwargs):
+        self.report_path = report_path
+        self.report_format = report_format  # for now only support json
+        self.report_dict = report_dict or self._pull_in_report()
+
+    def _pull_in_report(self):
+        if self.report_format == 'json':
+            try:
+                with open(self.report_path) as json_file:
+                    return json.load(json_file)
+            except IOError:
+                logger.exception("File {} does not exist".format(self.report_path))
+                raise
+
+    def parse_report_per_user(self, filter_by='failed', user=''):
+        tests = self.report_dict['tests']
+        assignee_wise_results = {}
+        for test_case in tests:
+            try:
+                result = tests[test_case]['statuses']['overall']
+            except KeyError:
+                # continue loop if any test is missing the statuses/overall
+                continue
+            # if user='' it will fetch for all users
+            if (result == filter_by and user in tests[test_case]['assignee']):
+                try:
+                    assignee_wise_results[tests[test_case]['assignee']].update(
+                        {test_case: result})
+                except KeyError:
+                    assignee_wise_results[tests[test_case]['assignee']] = (
+                        {test_case: result})
+        return assignee_wise_results
+
+
+@click.group("blame", help="Functions for generating reports on Tests grouped by user")
+def main():
+    pass
+
+
+@main.command("test-report", help="Returns a Dictionary containing failed test cases per user")
+@click.option("--report-file", help="Path to the file with json data")
+@click.option("--email-config-yaml", help="If you decide to use this option,copy yaml file similar"
+    " to report_parser_email_conf.yaml.template with correct values and pass its path."
+    " No email would be sent if --user is not set.",
+    type=click.Path())
+@click.option("--user", help="Set this if you want to filter the report for a specific user",
+    default='')
+@click.option("--filter-by", type=click.Choice(['passed', 'failed', 'skipped']),
+    help="Defaults to 'failed'", default='failed')
+def test_results_per_assignee(report_file, email_config_yaml, user, filter_by):
+    rp = ReportParser(report_file)
+    test_report_dict = rp.parse_report_per_user(filter_by, user)
+    # user is required before following routine would execute, else no email is sent
+    if test_report_dict and email_config_yaml and user:
```

Comment:
Is there anyway to generate the report for all users? Because if so I'd expect the default to do that.

Reply:
So by default user is `''` so it fetches report for each user. And I am adding a `default_to_addr`  so that if user is not defined, email would go to `default_to_addr`.  Currently I derive email from username, but if we want to be explicit that would require more work. I understood the task as we will have our kerberos username in assignee.